adminguide.rst

```diff
@@ -0,0 +1,514 @@
+
+.. _draining-resources:
+
+^^^^^^^^^^^^^^^^^^
+Draining Resources
+^^^^^^^^^^^^^^^^^^
+
+Resources may be temporarily removed from scheduling via the
+``flux resource drain`` command. Currently, resources may only be
+drained at the granularity of rank, for example:
+
+.. code-block:: console
+
+ $ sudo flux resource drain 4
+ $ sudo flux resource list
+      STATE NNODES   NCORES    NGPUS
+       free     15       30        0
+  allocated      0        0        0
+       down      1        2        0
+
+
+To return drained resources use ``flux resource undrain``:
+
+
+.. code-block:: console
+
+ $ sudo flux resource undrain 4
+ $ sudo flux resource list
+      STATE NNODES   NCORES    NGPUS
+       free     16       32        0
+  allocated      0        0        0
+       down      0        0        0
+
+
+.. _queue-admin:
+
+^^^^^^^^^^^^^^^^^^^^^^^
+Managing the Flux Queue
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The queue of jobs is managed by the flux job-manager, which in turn
+makes allocation requests for jobs in priority order to the scheduler.
+This queue can be managed using the ``flux-queue`` command.
+
+.. code-block:: console
+
+ Usage: flux-queue [OPTIONS] COMMAND ARGS
+   -h, --help             Display this message.
+
+ Common commands from flux-queue:
+    enable          Enable job submission
+    disable         Disable job submission
+    start           Start scheduling
+    stop            Stop scheduling
+    status          Get queue status
+    drain           Wait for queue to become empty.
+    idle            Wait for queue to become idle.
+
+
+~~~~~~~~~~~~~~~~~~~~~~~~
+Disabling job submission
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, the queue is *enabled*, meaning that jobs can be submitted
+into the system. To disable job submission, e..g to prepare the system
+for a shutdown, use ``flux queue disable``. To restore queue access
+use ``flux queue enable``.
+
+~~~~~~~~~~~~~~~~~~~~~~~
+Stopping job allocation
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The queue may also be stopped with ``flux queue stop``, which disables
+further allocation requests from the job-manager to the scheduler. This
+allows jobs to be submitted, but stops new jobs from being scheduled.
+To restore scheduling use ``flux queue start``.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Flux queue idle and drain
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``flux queue drain`` and ``flux queue idle`` commands can be used
+to wait for the queue to enter a given state. This may be useful when
+preparing the system for a downtime.
+
+The queue is considered *drained* when there are no more active jobs.
+That is, all jobs have completed and there are no pending jobs.
+``flux queue drain`` is most useful when the queue is *disabled* .
+
+The queue is "idle" when there are no jobs in the RUN or CLEANUP state.
+In the *idle* state, jobs may still be pending. ``flux queue idle``
+is most useful when the queue is *stopped*.
+
+To query the current status of the queue use the ``flux queue status``
```

Comment:
Just a thought but maybe we should encourage the use of "idle" for shutting down the system instance, since disallowing new job submissions might cause a running job that submits another job to fail?  

At least it might be helpful to note that use case for jobs, which motivated the `flux queue idle` method of waiting for the compute nodes to become idle.